specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-07-02-preview/managedClusters.json

```diff
@@ -6863,6 +6949,73 @@
       ],
       "description": "The OS option profile."
     },
+    "GuardrailsAvailableVersionsProperties": {
+      "type": "object",
+      "properties": {
+        "isDefaultVersion": {
+          "type": "boolean",
+          "readOnly": true
+        },
+        "support": {
+          "type": "string",
+          "enum": [
+            "Preview",
+            "Stable"
+          ],
+          "x-ms-enum": {
+            "name": "GuardrailsSupport",
+            "modelAsString": true,
+            "values": [
+              {
+                "value": "Preview",
+                "description": "The version is preview. It is not recommended to use preview versions on critical production clusters. The preview version may not support all use-cases."
+              },
+              {
+                "value": "Stable",
+                "description": "The version is stable and can be used on critical production clusters."
+              }
+            ]
+          },
+          "description": "Whether the version is preview or stable.",
+          "readOnly": true
+        }
+      }
+    },
+    "GuardrailsAvailableVersion": {
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/SubResource"
```

Comment:
To solve the `systemdata` linter warning, you can refer to ` "$ref": "../../../../../../common-types/resource-management/v3/types.json#/definitions/Resource` rather than `SubResource`. The only difference is that the common type has `systemData` too.

We aren't returning that right now in GET, but we probably could in the future.

Reply:
That makes sense. I made the change.